### PR TITLE
Feature/21 kerberos authentication

### LINKED ISSUE #21: Authentication: Kerberos
Connect to a datasource via Kerberos.

### PR DESCRIPTION
**Pull request by pj-spoelders**
Adds kerberos authentication (in its current form) 
Closes #21

### CODE REVIEW

File: UIAutomationTests/TestsAuthenticationWindowsKerberos.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+            var (error, grid) = testFixture.Test(sbPqFileStr, UIAutomationTestFixture.AuthenticationMethod.None);
+
+            Assert.True(String.IsNullOrWhiteSpace(error), $@"Errormessage: {error}");
+
+            string connectionStrKerberosServicenameField = "KERBEROSSERVICENAME";
+            //assert there's a row with a Kerberos service record field
+            int? kerberosServiceNameRecordRow = Utilities.FindRow(grid, 0, connectionStrKerberosServicenameField);
+            Assert.True(kerberosServiceNameRecordRow.HasValue, "KERBEROSSERVICENAME is missing from the Connection String Record.");
+            Assert.True(grid.Rows[kerberosServiceNameRecordRow.Value].Cells[1].Value == "exasol/" + dnsRecord);
```
Comment (chiaradiamarcelo) [inline UIAutomationTests/TestsAuthenticationWindowsKerberos.cs:31]: maybe we can extract `grid.Rows[kerberosServiceNameRecordRow.Value].Cells[1].Value` to a `getConnectionString()` method or something similar

File: UIAutomationTests/TestsGeneralFunctionality.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ public void OdbcDatasourceOverview()
         }
         //metadata test - adventure works (populated schema)
         [Fact]
-        public void OdbcDatasourcePopulatedSchemaAdventureWorksRows()
+        public void OdbcDatasourcePopulatedSchemaRows()
```
Comment (chiaradiamarcelo) [inline UIAutomationTests/TestsGeneralFunctionality.cs:35]:
```suggestion
        public void TestOdbcDatasourcePopulatedSchemaRows()
```

and same with other tests

**Review by chiaradiamarcelo — APPROVED** [review]

File: UIAutomationTests/TestsAuthenticationWindowsKerberos.cs (hunk 1 above)
Comment (pj-spoelders) [reply UIAutomationTests/TestsAuthenticationWindowsKerberos.cs:31]: Will do if more work on it is performed in the future.

File: UIAutomationTests/UIAutomationTestFixture.cs [hunk 3]
```diff
@@ -160,20 +161,19 @@ private void EnterCredentialsKeyOIDCToken()
             buttons[0].AsButton().Invoke();
 
         }
-        private void RemoveCredentials()
+        private void RemoveCredentials(UIAutomationTestFixture.AuthenticationMethod authenticationMethod)
```
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply UIAutomationTests/UIAutomationTestFixture.cs:164]: Will fix in next PR

**Pull request merged**